jdaviz/configs/cubeviz/plugins/tests/test_cubeviz_aperphot.py

```diff
@@ -113,13 +124,85 @@ def test_cubeviz_aperphot_generated_3d_gaussian_smooth(cubeviz_helper, image_cub
+    # should be converted to a surface brighntess after loading), flux per sr
+    # and flux per square pixel. the pixel area for per steraidan cubes is
+    # set so the output values between units will be the same
+
+    cube = spectrum1d_cube_custom_fluxunit(fluxunit=cube_unit)
+    cubeviz_helper.load_data(cube, data_label="test")
+
+    aper = RectanglePixelRegion(center=PixCoord(x=3, y=1), width=1, height=1)
+    bg = RectanglePixelRegion(center=PixCoord(x=2, y=0), width=1, height=1)
+    cubeviz_helper.load_regions([aper, bg])
+
+    plg = cubeviz_helper.plugins["Aperture Photometry"]._obj
+    plg.dataset_selected = "test[FLUX]"
+    plg.aperture_selected = "Subset 1"
+    plg.background_selected = "Subset 2"
+
+    #  Check that the default flux scaling is present for MJy / sr cubes
+    if cube_unit == (u.MJy / u.sr):
+        assert_allclose(plg.flux_scaling, 0.003631)
+    else:
+        assert_allclose(plg.flux_scaling, 0.0)
+
+    # if cube is MJy / sr, set pixel area to 1 sr / pix2 so
+    # we can directly compare outputs between per sr and per pixel cubes, which
+    # will give the same results with this scaling
+    if cube_unit == (u.MJy / u.sr):
+        assert_allclose(plg.pixel_area, 0.01)  # check default
+        plg.pixel_area = 1 * (u.sr).to(u.arcsec*u.arcsec)
+        solid_angle_unit = u.sr
+
+    else:
+        # for per pixel cubes, set flux scaling to default for MJy / sr cubes
+        # so we can directly compare. this shouldn't be populated automatically,
+        # which is checked above
+        plg.flux_scaling = 0.003631
+        solid_angle_unit = u.pix * u.pix
+        cube_unit = u.MJy / solid_angle_unit  # cube unit in app is now per pix2
+
+    plg.vue_do_aper_phot()
+    row = cubeviz_helper.get_aperture_photometry_results()[0]
+
+    # Basically, we should recover the input rectangle here, minus background.
+    assert_allclose(row["xcenter"], 3 * u.pix)
+    assert_allclose(row["ycenter"], 1 * u.pix)
+    # (15 - 10) MJy/sr x 1 sr, will always be MJy since solid angle is multiplied out
+    assert_allclose(row["sum"], 5.0 * u.MJy)
+
+    assert_allclose(row["sum_aper_area"], 1 * (u.pix * u.pix))
+
+    #  we forced area to be one sr so MJy / sr and MJy / pix2 gave the same result
+    assert_allclose(row["pixarea_tot"], 1.0 * solid_angle_unit)
+
+    # also forced flux_scaling to be the same for MJy / sr cubes, which get a
+    # default value populated, and MJy / pix2 which have a default of 0.0
+    assert_allclose(row["aperture_sum_mag"], -7.847359 * u.mag)
+
+    assert_allclose(row["mean"], 5 * (cube_unit))
+    # TODO: check if slice plugin has value_unit set correctly
+    assert_quantity_allclose(row["slice_wave"], 0.46236 * u.um)
+
+
 def test_cubeviz_aperphot_cube_orig_flux_mjysr(cubeviz_helper, spectrum1d_cube_custom_fluxunit):
+    # this test is essentially the same as test_cubeviz_aperphot_cube_sr_and_pix2 but for a single
```

Comment:
Or maybe you don't have to parametrize this particular unit in `test_cubeviz_aperphot_cube_sr_and_pix2`, then you don't have duplicate test case?

Reply:
I think it's important to test flux/sr and flux/pix2 give the same results with the proper pixel area factor, which is what parameterizing this does